File: src/main/java/testla/screenplay/task/Task.java
package testla.screenplay.task;

import testla.screenplay.actor.IActor;
import testla.screenplay.task.ITask;

/**
 * TODO: Add Description
 *
 * @author Patrick Doering
 */
public abstract class Task implements ITask {

    public abstract Object performAs(IActor actor);
}

File: src/main/kotlin/de/p7s1/testla/screenplay/Task.kt
package de.p7s1.testla.screenplay
import de.p7s1.testla.IActor
import de.p7s1.testla.ITask

abstract class Task: ITask {
    /**
     *  Makes the provided IActor perform this Action.
     *
     * @param actor the actor
     * @returns Any
     *
     * @override This method will have to be overridden with the actual integration of an action.
     */
    abstract override fun performAs(actor: IActor): Any
}